### PR TITLE
[Auth] Remove the `.showLoginEmail` segue and navigate programmatically

### PR DESCRIPTION
**Pull request by mindgraffiti**
Ref. #2003.

### In this PR
- TIL that Cocoapods does not like our beta versions and will ignore newer beta versions, so I made the beta versions explicit in the podfile.
- Authenticator has been updated to remove the storyboard segue `.showLoginEmail`
- See also https://github.com/wordpress-mobile/WordPressAuthenticator-iOS/pull/207

### To Test
1. Check out this branch
2. `rake dependencies`
3. Run unit tests
4. Build and run
5. Log out
6. Select the "Log in with Jetpack" button. 

### Do not merge until
1. The Authenticator PR has been merged: wordpress-mobile/WordPressAuthenticator-iOS#207 ✔️
2. A new Authenticator release has been created ✔️
3. The release is published on Cocoapods trunk ✔️
4. This PR points to the new release ✔️

**Expected result:** it should show the expected LoginEmailViewController

Update release notes:

- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

**Comment by shiki** [comment]
@mindgraffiti It looks like there are conflicts in `Podfile` and `Podfile.lock`. 🙂

**Comment by mindgraffiti** [comment]
@shiki fixed 🙂

File: Podfile [hunk 1]
```diff
@@ -20,18 +20,18 @@ target 'WooCommerce' do
...
+  pod 'Automattic-Tracks-iOS', '~> 0.4.4-beta.2'
 
-  pod 'Gridicons', '~> 0.20-beta'
+  pod 'Gridicons', '~> 0.20-beta.1'
 
   # To allow pod to pick up beta versions use -beta. E.g., 1.1.7-beta.1
-  #pod 'WordPressAuthenticator', :git => 'https://github.com/wordpress-mobile/WordPressAuthenticator-iOS.git', :branch => 'task/wc-support-site-url-login'
-  pod 'WordPressAuthenticator', '~> 1.11.0-beta.13'
+  pod 'WordPressAuthenticator', :git => 'https://github.com/wordpress-mobile/WordPressAuthenticator-iOS.git', :branch => 'issue/204-remove-showLoginEmail'
+  #pod 'WordPressAuthenticator', '~> 1.11.0-beta.14'
```
Comment (shiki) [inline Podfile]: We will be changing this back to the `beta` version later when https://github.com/wordpress-mobile/WordPressAuthenticator-iOS/pull/207 is merged, right?

File: Podfile [hunk 2]
```diff
@@ -20,18 +20,18 @@ target 'WooCommerce' do
 
   # Use the latest bugfix for coretelephony
   #pod 'Automattic-Tracks-iOS', :git => 'https://github.com/Automattic/Automattic-Tracks-iOS.git', :tag => '0.2.4-beta.1'
-  pod 'Automattic-Tracks-iOS', '~> 0.4.4-beta'
+  pod 'Automattic-Tracks-iOS', '~> 0.4.4-beta.2'
```
Comment (shiki) [inline Podfile]: > TIL that Cocoapods does not like our beta versions and will ignore newer beta versions, so I made the beta versions explicit in the podfile.

That's weird. I changed this back to:

```ruby
  pod 'Automattic-Tracks-iOS', '~> 0.4.4-beta'
```

And then executed: 

```shell
$ rm -rf Pods Podfile.lock
$ bundle exec pod install
```

... and Cocoapods **successfully** fetched the `-beta.2`:

```
Analyzing dependencies
Downloading dependencies
Installing 1PasswordExtension (1.8.6)
Installing Alamofire (4.8.0)
Installing Automattic-Tracks-iOS (0.4.4-beta.2) <----
```

The generated `Podfile.lock` is also pointing to `-beta.2`: 

```
PODS:
  - 1PasswordExtension (1.8.6)
  - Alamofire (4.8.0)
  - Automattic-Tracks-iOS (0.4.4-beta.2): <---
    - CocoaLumberjack (~> 3.5.2)
    - Reachability (~> 3.1)
    - Sentry (~> 4)
    - UIDeviceIdentifier (~> 1)
```

I then checked `TracksConstants.m` and it has this:

```objc
NSString *const TracksLibraryVersion = @"0.4.4-beta.2";
```

I have a feeling I misunderstood the problem. 😄 What am I missing? 🤔

File: RELEASE-NOTES.txt [hunk 3]
```diff
@@ -3,6 +3,7 @@
 - bugfix: now in the Order List the order status label is no more clipped
 - bugfix: now the launch screen is no more stretched
 - The Shipping Provider flow, will be called now Shipping Carrier.
+- [internal] The signup and login flows have code changes. See https://git.io/Jv1Me for testing details.
```
Comment (shiki) [inline RELEASE-NOTES.txt]: I like this! 💜

File: Podfile (hunk 1 above)
Comment (mindgraffiti) [reply Podfile]: Correct! Please excuse the oversight. I updated the PR description with the `do not merge until` section.

Comment (shiki) [reply Podfile]: No worries. I wasn't sure. It's been a while since I changed one of the pods. 😄

File: Podfile (hunk 2 above)
Comment (mindgraffiti) [reply Podfile]: 🤔 I wonder if the problem is only happening on WPiOS. I'll get back to you on that. I need to ask Jeremy some questions 😄

Comment (mindgraffiti) [reply Podfile]: Okay, I don't yet know why, but I can confirm this is only happening on WPiOS. Will remove 🙂

**Comment by mindgraffiti** [comment]
Thank you @shiki !

**Pull request merged**